src/components/header.tsx

```diff
@@ -2,14 +2,19 @@ import { Link } from 'gatsby';
 import React from 'react';
 import logoLight from '../images/logos/nodejs-logo-light-mode.svg';
 import logoDark from '../images/logos/nodejs-logo-dark-mode.svg';
+import DarkModeController from '../util/DarkModeController';
 
 const activeStyleTab = {
   fontWeight: 'var(--font-weight-semibold)',
   color: 'var(--color-text-accent)',
   borderBottom: 'var(--space-04) inset var(--color-text-accent)',
 };
 
-const Header = () => (
+interface Props {
+  darkModeController?: DarkModeController;
```

Comment:
Alt: Just have the header import the controller. May require structural changes to the controller code to not maintain its own state and instead run off local storage / user preferences. This way, most of those helper methods can exist on the module export as pure functions instead of requiring a new instance to be made each time.